### PR TITLE
feat(textbox): add prefix prop to display emphasized prefix (FE-2954)

### LINKED ISSUE #3078: Textbox prefix
## Feature request

Allow the user to enter a prefix in this case `GB` that the user will no be able to enter/edit.

### Current behaviour

image

### Desired behaviour

image

### Suggested solution
N/A

### Describe alternatives you've considered
N/A

### Additional context

Required for CM

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| carbon-react     | 35.1.1 |

### PR DESCRIPTION
**Pull request by grabkowski**
### Proposed behaviour
Screenshot 2020-08-28 at 13 18 26

Allow standard inputs to have the ability of a prefix in bold by providing a prop that accepts a string.

### Current behaviour

No prefix prop available. 

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
~~- [ ] All themes are supported if required~~
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Additional context
closes #3078

`input` component user agent padding removed to set proper margin
removed static height of the input component

### Testing instructions
1. npm start
2. http://localhost:9001/?path=/story/experimental-textbox--default

### CODE REVIEW

**Review by DlgSHi — COMMENTED** [review]
If there is no any story for `chromatic` - `cypress` tests need to be added for the prefix

File: src/__experimental__/components/textbox/__internal__/prefix.style.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import styled from 'styled-components';
+
+const StyledPrefix = styled.span`
+  align-self: center;
+  font-weight: 900;
+  margin-right: 4px;
```
Comment (jamime) [inline src/__experimental__/components/textbox/__internal__/prefix.style.js:6]: Spec says this should be 8px.

Comment (jamime) [inline src/__experimental__/components/textbox/__internal__/prefix.style.js:6]: Spec says this gap should be 8px, should this be changed?

Comment (grabkowski) [reply src/__experimental__/components/textbox/__internal__/prefix.style.js:6]: input already has it's own padding, so it should add up to 8px gap

**Review by sjohnson-sage — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by sjohnson-sage — DISMISSED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by DlgSHi — DISMISSED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Review by dawidzarzycki — APPROVED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Review by sjohnson-sage — DISMISSED** [review]

**Review by dawidzarzycki — APPROVED** [review]

**Review by dawidzarzycki — APPROVED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Pull request merged**